Sources/StreamChat/Models/Channel.swift

```diff
@@ -602,3 +646,164 @@ public extension ChatChannel {
+            truncatedAt: truncatedAt,
+            isHidden: false,
+            createdBy: createdBy?.asModel(),
+            config: config,
+            ownCapabilities: Set(ownCapabilities?.compactMap { ChannelCapability(rawValue: $0) } ?? []),
+            isFrozen: isFrozen,
+            isDisabled: isDisabled,
+            isBlocked: isBlocked ?? false,
+            lastActiveMembers: members?.compactMap { $0.asModel(channelId: cid) } ?? [],
+            membership: nil,
+            currentlyTypingUsers: [],
+            lastActiveWatchers: [],
+            team: team,
+            unreadCount: ChannelUnreadCount(messages: 0, mentions: 0),
+            watcherCount: 0,
+            memberCount: memberCount,
+            reads: [],
+            cooldownDuration: cooldownDuration,
+            extraData: extraData,
+            latestMessages: [],
+            lastMessageFromCurrentUser: nil,
+            pinnedMessages: [],
+            muteDetails: nil,
+            previewMessage: nil,
+            draftMessage: nil,
+            activeLiveLocations: []
+        )
+    }
+}
+
+extension MemberPayload {
+    /// Converts the MemberPayload to a ChatChannelMember model
+    /// - Parameter channelId: The channel ID the member belongs to
+    /// - Returns: A ChatChannelMember instance, or nil if user is missing
+    func asModel(channelId: ChannelId) -> ChatChannelMember? {
+        guard let userPayload = user else { return nil }
+        let user = userPayload.asModel()
+        
+        return ChatChannelMember(
+            id: user.id,
+            name: user.name,
+            imageURL: user.imageURL,
+            isOnline: user.isOnline,
+            isBanned: user.isBanned,
+            isFlaggedByCurrentUser: user.isFlaggedByCurrentUser,
+            userRole: user.userRole,
+            teamsRole: user.teamsRole,
+            userCreatedAt: user.userCreatedAt,
+            userUpdatedAt: user.userUpdatedAt,
+            deactivatedAt: user.userDeactivatedAt,
+            lastActiveAt: user.lastActiveAt,
+            teams: user.teams,
+            language: user.language,
+            extraData: user.extraData,
+            memberRole: MemberRole(rawValue: role?.rawValue ?? "member"),
+            memberCreatedAt: createdAt,
+            memberUpdatedAt: updatedAt,
+            isInvited: isInvited ?? false,
+            inviteAcceptedAt: inviteAcceptedAt,
+            inviteRejectedAt: inviteRejectedAt,
+            archivedAt: archivedAt,
+            pinnedAt: pinnedAt,
+            isBannedFromChannel: isBanned ?? false,
+            banExpiresAt: banExpiresAt,
+            isShadowBannedFromChannel: isShadowBanned ?? false,
+            notificationsMuted: notificationsMuted,
+            memberExtraData: [:]
+        )
+    }
+}
+
+extension ChannelReadPayload {
+    /// Converts the ChannelReadPayload to a ChatChannelRead model
+    /// - Returns: A ChatChannelRead instance
+    func asModel() -> ChatChannelRead {
```

Comment:
maybe instead we always return an empty array for reads?

Reply:
We can keep this for now. It might be useful when needing to map channels to payloads for other things. Since we are doing this, better to try mapping all the data we can